stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -56,7 +60,13 @@ public ResponseBodyData readFile(String requestId) throws IOException {
       }
       ResponseBodyData bodyData = new ResponseBodyData();
       bodyData.base64Encoded = firstByte != 0;
-      bodyData.data = readContentsAsUTF8(in);
+      if (mRequestIdMap.containsKey(requestId)) {
+        AsyncPrettyPrinter asyncPrettyPrinter = mRequestIdMap.get(requestId);
+        bodyData.data = prettyPrintContent(in, asyncPrettyPrinter);
```

Comment:
We have a problem here.  This function can block for arbitrary lengths of time and in fact can time out.  Your code doesn't try to address that at all and could just wedge the UI.

Please consider a timeout here and also a test to see what the UI will do in the event that no response is given.